app/models/external_upload_stub.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require "digest/sha1"
+
+class ExternalUploadStub < ActiveRecord::Base
+  CREATED_EXPIRY_HOURS = 1
+  UPLOADED_EXPIRY_HOURS = 24
+
+  belongs_to :created_by, class_name: 'User'
+
+  scope :expired_created, -> {
+    where(
+      "status = ? AND created_at <= ?",
+      ExternalUploadStub.statuses[:created],
+      CREATED_EXPIRY_HOURS.hours.ago
+    )
+  }
+
+  scope :expired_uploaded, -> {
+    where(
+      "status = ? AND created_at <= ?",
+      ExternalUploadStub.statuses[:uploaded],
+      UPLOADED_EXPIRY_HOURS.hours.ago
+    )
+  }
+
+  before_create do
+    self.unique_identifier = SecureRandom.hex(20)
+    self.status = ExternalUploadStub.statuses[:created] if self.status.blank?
+  end
+
+  def self.statuses
+    @statuses ||= Enum.new(
+      created: 1,
+      uploaded: 2,
+      failed: 3
+    )
+  end
+
+  # TODO (martin): Lifecycle rule would be best to clean stuff up in the external
+  # systems, I don't think we really want to be calling out to the external systems
+  # here right?
+  def self.cleanup!
```

Comment:
Not quite sure whether I should put a lifecycle rule in this same commit; no one is using this yet and it's all behind experimental site settings. When I do add one it will be done in a similar way to the tombstone lifecycle rule, but for the `temp` folder this time:

https://github.com/discourse/discourse/blob/e1b0f67240b2ee7b297bcca6fec486678d156a2b/lib/s3_helper.rb#L191-L195

Reply:
Yup I agree we can follow up with this in a separate commit.